src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8493,175 +8499,83 @@ protected MethodGroupResolution ResolveExtension(
-                    // first non-applicable result and continue searching for an applicable result.
-                    if (extensionMethodResult.HasAnyApplicableMethod)
-                    {
-                        firstResult.Free();
-                        actualArguments?.Free();
-                        return extensionMethodResult;
-                    }
-                    else if (firstResult.IsEmpty)
-                    {
-                        firstResult = extensionMethodResult;
-                    }
-                    else
-                    {
-                        // Neither the first result, nor applicable. No need to save result.
-                        extensionMethodResult.Free();
-                    }
-                }
+                lookupOptions |= LookupOptions.MustBeInvocableIfMember;
             }
 
-            Debug.Assert((actualArguments == null) || !firstResult.IsEmpty);
-            actualArguments?.Free();
-
-            return firstResult;
+            var diagnostics = BindingDiagnosticBag.GetInstance(withDiagnostics: true, withDependencies);
+            CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = this.GetNewCompoundUseSiteInfo(diagnostics);
 
-            static bool tryResolveExtensionMember(Binder binder, SyntaxNode expression, string memberName, AnalyzedArguments? analyzedArguments, BoundExpression left,
-                ImmutableArray<TypeWithAnnotations> typeArgumentsWithAnnotations, OverloadResolution.Options options, RefKind returnRefKind, in CallingConventionInfo callingConvention,
-                TypeSymbol? returnType, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo, ExtensionScope scope, out MethodGroupResolution extensionResult)
+            foreach (var scope in new ExtensionScopes(this))
             {
-                var lookupResult = LookupResult.GetInstance();
-
-                int arity = typeArgumentsWithAnnotations.IsDefault ? 0 : typeArgumentsWithAnnotations.Length;
-                var lookupOptions = (arity == 0) ? LookupOptions.AllMethodsOnArityZero : LookupOptions.Default;
-                if (analyzedArguments is not null)
-                {
-                    lookupOptions |= LookupOptions.MustBeInvocableIfMember;
-                }
+                lookupResult.Clear();
+                classicExtensionLookupResult.Clear();
 
                 scope.Binder.LookupExtensionMembersInSingleBinder(
                     lookupResult, left.Type!, memberName, arity,
-                    basesBeingResolved: null, lookupOptions, originalBinder: binder, ref useSiteInfo);
+                    basesBeingResolved: null, lookupOptions, originalBinder: this, ref useSiteInfo);
+
+                this.LookupExtensionMethods(classicExtensionLookupResult, scope, memberName, arity, ref useSiteInfo);
+                diagnostics.Add(expression, useSiteInfo);
```

Comment:
>useSiteInfo [](http://example.com/codeflow?start=44&length=11)

It looks like we might be adding the information already added during the previous iteration